### DISCUSSION TITLE
What happens if two different modules receive transmit()?

### DISCUSSION THREAD

**Discussion by timmornYE** [jgromes/RadioLib#761] [General]
I miss some information in the API what happens on `transmit (const char *  str,uint8_t  addr = 0 )`  if two nodes receive it concurrently? Because receive() is blocking I presume that it unblocks as soon as the first node received it, so data will not arrive in node 2. But this way I see a problem because I can never know if someone installs a concurrent receiver near my location?

**Comment by jgromes** [comment]
How would the receiver A "know" there's another receiver it should be waiting for ... ? You can of course have multiple receivers. In fact you can never know who is receiveing your data. It's like talking  to two people at once, it doesn't mean the second one cannot hear you if the first one can.

**Reply by timmornYE** [reply]
Thank you for your answer. This point is clear to me. My question was more lib specific. It is no problem to me if someone other receives the package (think this is possible because of the long range), but I do not want that my receiver does not get the package. So I wondered if transmit() stops sending if an other receiver receives the package earlier or is it more this way that transmit() sends a fixed amount of time and stops no matter if 1 or 10 receiver received the package?

**Reply by jgromes** [reply]
`transmit()` does exactly what it says, it transmits the data - once, and it does not check whether it was received by anyone. So by its own, it does not guarantee message was delivered, completely and without errors. If you want something like that, you will either have to find an existing protocol that does this, or implement it yourself (e.g. using acknowledge messages, node address etc. etc.)

**Reply by timmornYE** [reply]
Thanks that helps a lot. Because you write "blocking" I thought the lib has some form of ack.

**Reply by jgromes** [reply]
The term "blocking" has a defined meaning in software engineering - that it blocks execution until completion. Completion is when all data is sent, not sent and its reception verified.
